docs/json-cadence-spec.md

```diff
@@ -366,3 +370,456 @@ Composite fields are encoded as a list of name-value pairs in the order in which
   }
 }
 ```
+
+---
+
+# Types
+
+## Simple Types
+
+These are basic types like `Int`, `String`, or `StoragePath`. 
+
+```json
+{
+  "kind": "<type>"
+}
+```
+
+### Example
+
+```json
+{
+  "kind": "UInt8"
+}
+```
+
+---
+
+## Optional Types
+
+```json
+{
+  "kind": "Optional",
+  "type": <type>
+}
+```
+
+### Example 
+
+```json
+{
+  "kind": "Optional",
+  "type": {
+    "kind": "String"
+  }
+}
+```
+
+---
+
+## Variable Sized Array Types
+
+```json
+{
+  "kind": "VariableSizedArray",
+  "type": <type>
+}
+```
+
+### Example 
+
+```json
+{
+  "kind": "VariableSizedArray",
+  "type": {
+    "kind": "String"
+  }
+}
+```
+
+---
+
+## Constant Sized Array Types
+
+```json
+{
+  "kind": "ConstantSizedArray",
+  "type": <type>,
+  "size": "<length of array>",
+}
+```
+
+### Example 
+
+```json
+{
+  "kind": "ConstantSizedArray",
+  "type": {
+    "kind": "String"
+  },
+  "size":"3"
+}
+```
+
+---
+
+## Dictionary Types
+
+```json
+{
+  "kind": "Dictionary",
+  "key": <type>,
+  "value": <type>
+}
+```
+
+### Example 
+
+```json
+{
+  "kind": "Dictionary",
+  "key": {
+    "kind": "String"
+  }, 
+  "value": {
+    "kind": "UInt16"
+  }, 
+}
+```
+
+---
+
+## Composite Types
+
+```json
+{
+  "kind": "Struct" | "Resource" | "Event" | "Constract" | "StructInterface" | "ResourceInterface" | "ContractInterface",
+  "type": "",
```

Comment:
Is "type" field needed?

Reply:
Yes, because Enums have a type but also must have the same structure as the other composites, non-enum composites were implemented to have an empty type field.